README.md

```diff
@@ -16,90 +16,192 @@ and to keep them as similar as possible:
+environment as much as possible.
+
+#### Privileged Mode
+
+In privileged mode the rpm-ostree compose uses container features
+itslef, thus requires a privileged container in order to perform the
+compose. This is known as [recursive containers](https://github.com/projectatomic/bubblewrap/issues/284).
+
+### Container Build
 ---
 
-Here we store data in `/srv/coreos` on our host system.  You can choose
-any directory you like.  You should run these commands as `root`.
+To completely rebuild the coreos-assembler container image locally, execute
+`$ sudo podman build -t localhost/coreos-assembler .` or
+`$ sudo podman build -t localhost/coreos-assembler -f Dockerfile.rhel .`
+from the `coreos-assembler` repository. If building the RHEL version please 
+note you will need a maipo directory with references to
+proper repositories.
+
+You can upload your built image to a registry such as quay.io by doing the following:
 
 ```
-$ mkdir /srv/coreos
-$ cd /srv/coreos
-$ alias coreos-assembler='podman run --rm -ti --privileged --userns=host -v $(pwd):/srv --workdir /srv quay.io/coreos-assembler/coreos-assembler'
+$ sudo podman push <image id> example.com/<account name>/coreos-assembler
 ```
 
-If you need access to CA certificates on your host (for example, when you need to access
-a git repo that is not on the public Internet), you can mount in the host certificates
-as read-only.  For example, on a Fedora host the alias would change to:
+### Setup
+---
 
-`$ alias coreos-assembler='podman run --rm -ti --privileged --userns=host -v /etc/pki:/etc/pki:ro -v $(pwd):/srv --workdir /srv quay.io/coreos-assembler/coreos-assembler'`
+Let's set up our working directory first. We'll use `/srv/coreos` on
+our host system. You can choose any directory you like. 
 
-See this [Stack Overflow question](https://stackoverflow.com/questions/26028971/docker-container-ssl-certificates) for additional discussion.
+```
+$ mkdir /srv/coreos && chmod 777 /srv/coreos && cd /srv/coreos
+$ chcon -Rt svirt_sandbox_file_t /srv/coreos  # set SELinux file contexts for sharing
+```
+
+Now we'll define a bash function that we can use to call the assembler
+container:
+
+
+```
+$ cosa() {
+    env | grep COREOS_ASSEMBLER
+    set -x # so we can see what command gets run
+    sudo podman run --rm -ti -v ${PWD}:/srv/ --userns=host --device /dev/kvm --name cosa            \
+               ${COREOS_ASSEMBLER_PRIVILEGED:+--privileged}                                         \
+               ${COREOS_ASSEMBLER_CONFIG_GIT:+-v $COREOS_ASSEMBLER_CONFIG_GIT:/srv/src/config/:ro}  \
+               ${COREOS_ASSEMBLER_GIT:+-v $COREOS_ASSEMBLER_GIT/src/:/usr/lib/coreos-assembler/:ro} \
+               ${COREOS_ASSEMBLER_CONTAINER_RUNTIME_ARGS}                                           \
+               ${COREOS_ASSEMBLER_CONTAINER:-quay.io/coreos-assembler/coreos-assembler:latest} $@
+    set +x
+}
+```
+
+**NOTE**: I'm using `cosa` here as it is much easier to type than `coreos-assembler`.
```

Comment:
Oh, you did write this here.  I dunno...particularly inside the container it's `coreo<TAB>` looks like.